feat(core): add `@Transactional()` decorator (#6143)
mikro-orm/mikro-orm: docs/docs/transactions.md

}
```

> Another way to handle explicit transactions is by using `@Transactional()` decorator.

Reviewer: ```suggestion
Another way to handle explicit transactions is by using the`@Transactional()` decorator.
```

```ts
import { EntityManager, MikroORM, Transactional } from '@mikro-orm/core';

export class MyService {

  constructor(private readonly em: EntityManager) { }

  @Transactional()
  async doSomething() {
    //... do some work
    const user = new User(...);
    user.name = 'George';
    em.persist(user); 
  }

}
```

This decorator wraps the method with `em.transactional()`, so you can provide `TransactionOptions` just like with `em.transactional()`. The difference is that you can specify the context in which the transaction begins by providing `context` option, and if omitted, the transaction will begin in the current context implicitly. It works on async functions and can be nested with `em.transactional()`.

Explicit transaction demarcation is required when you want to include custom DBAL operations in a unit of work or when you want to make use of some methods of the EntityManager API that require an active transaction. Such methods will throw a `ValidationError` to inform you of that requirement.

> `em.transactional(cb)` and `@Transactional()` will flush the inner `EntityManager` prior to transaction commit.

Reviewer: ```suggestion
`em.transactional(cb)` and the `@Transactional()` will flush the inner `EntityManager` before transaction commit.
```
